src/SupportMessage/SupportMessage.tsx

```diff
@@ -62,6 +62,10 @@ export type SupportMessageProps = {
   className?: string
   description: string | ReactElement
   onClick?: MouseEventHandler
+  /**
+   * Right side content, usually an icon or a button
+   */
+  rightSideComponent?: ReactElement
```

Comment:
nit: Do we want to be bit more flexible and use `ReactNode` here? would mean we could use text without wrapping in a component. Would probably mean we want some default flex styling for the parent in these cases (we should probably add this anyway)

Reply:
The approach I try to have here is two things:
- A approach that favours generic composability over strict extensibility.
- Only adding functionality that is needed, avoiding changes that we're unsure will be used

I don't really have a use case for simply allowing text on the right side, I also have a question on what default props this text fallback would have, making the wrong choice might be more detrimental than beneficial.

On the flip-side, I understand this generic composability approach requires a little more code to take advantage of, which I would say would be the trade-off.